app/workers/mail_chimp_worker.rb

```diff
@@ -0,0 +1,12 @@
+class MailChimpWorker
+  include Sidekiq::Worker
+  include Mailchimp
+
+  def perform(user_id)
+    user= User.find(user_id)
+    subscribe_to_mailchimp(ENV['USERS_LIST_ID'], user)
+    if subscribe_newsletter
```

Comment:
Where is `subscribe_newsletter` defined ? If you are referring to the subscribe_newsletter attribute in user, then you have to call them by `user.subscribe_newsletter` . But it will always nil since we are fetching this record from db

Reply:
Good catch, @redtachyons! Also, it looks to me like we're calling `subscribe_to_newsletter` regardless . . . possibly a copy and paste mix up?

Could we resolve this discrepancy by only triggering this job if the user checks the checkbox?